include/grpcpp/security/auth_metadata_processor.h

```diff
@@ -27,29 +27,39 @@
 
 namespace grpc {
 
-/// Interface allowing custom server-side authorization based on credentials
-/// encoded in metadata.  Objects of this type can be passed to
-/// \a ServerCredentials::SetAuthMetadataProcessor().
+// Interface allowing custom server-side authorization based on credentials
+// encoded in metadata.  Objects of this type can be passed to
+// \a ServerCredentials::SetAuthMetadataProcessor().
+// Please also check out grpc::experimental::Interceptor for another way to do
+// customized operations on the information provided by a specific call.
 class AuthMetadataProcessor {
  public:
   typedef std::multimap<grpc::string_ref, grpc::string_ref> InputMetadata;
   typedef std::multimap<std::string, std::string> OutputMetadata;
 
   virtual ~AuthMetadataProcessor() {}
 
-  /// If this method returns true, the \a Process function will be scheduled in
-  /// a different thread from the one processing the call.
+  // If this method returns true, the \a Process function will be scheduled in
+  // a different thread from the one processing the call.
   virtual bool IsBlocking() const { return true; }
 
-  /// context is read/write: it contains the properties of the channel peer and
-  /// it is the job of the Process method to augment it with properties derived
-  /// from the passed-in auth_metadata.
-  /// consumed_auth_metadata needs to be filled with metadata that has been
-  /// consumed by the processor and will be removed from the call.
-  /// response_metadata is the metadata that will be sent as part of the
-  /// response.
-  /// If the return value is not Status::OK, the rpc call will be aborted with
-  /// the error code and error message sent back to the client.
+  // Processes a Call associated with a Channel.
+  // auth_metadata: the authentication metadata associated with the particular
+  //   call
+  // context: contains the channel-level info, e.g. the peer identity. This
+  //   parameter is readable and writable. Note that since the information is
+  //   shared for all calls associated with the channel, if we update the info
+  //   in a specific call, all the subsequent calls will see the updates. A
+  //   typical usage of context is to use |auth_metadata| to infer the peer
+  //   identity, and augment it with properties.
+  // consumed_auth_metadata: contains the metadata that users want to remove for
```

Comment:
This comment is incorrect.  The metadata specified here is not removed from subsequent calls; it is removed from the current call, so that the metadata is no longer present when the server application sees it.